code/datums/martial/buster_style.dm

```diff
@@ -0,0 +1,493 @@
+				target.visible_message(span_warning("[O] collides with [T]!"))
+			drop()
+			return
+		for(var/obj/Z in T.contents) // crash into something solid and damage it along with thrown objects that hit it
+			for(var/obj/O in thrown) 
+				if(Z.density == TRUE) 
+					O.take_damage(objdam) 
+					if(istype(O, /obj/mecha)) // mechs are probably heavy as hell so stop flying after making contact with resistance
+						thrown -= O
+			if(Z.density == TRUE && Z.anchored == FALSE) // if the thing hit isn't anchored it starts flying too
+				thrown |= Z 
+				Z.take_damage(50) 
+			if(Z.density == TRUE && Z.anchored == TRUE) // If the thing is solid and anchored like a window or grille or table it hurts people thrown that crash into it too
+				for(var/mob/living/S in thrown) 
+					grab(user, S, slamdam) 
+					S.Knockdown(1.5 SECONDS)
+					S.Immobilize(1.5 SECONDS)
+					if(isanimal(S) && S.stat == DEAD)
+						S.gib()
+					if(istype(Z, /obj/machinery/disposal/bin)) // dumpster living things tossed into the trash
+						var/obj/machinery/disposal/bin/dumpster = D
+						S.forceMove(Z)
+						Z.visible_message(span_warning("[S] is thrown down the trash chute!"))
+						dumpster.do_flush()
+						drop()
+						return
+				Z.take_damage(objdam)
+				if(Z.density == TRUE && Z.anchored == TRUE)
+					drop()
+					return // if the solid thing we hit doesnt break then the thrown thing is stopped
+		for(var/mob/living/M in T.contents) // if the thrown mass hits a person then they get tossed and hurt too along with people in the thrown mass
+			if(user != M)
+				grab(user, M, slamdam) 
+				M.Knockdown(1.5 SECONDS) 
+				for(var/mob/living/S in thrown)
+					grab(user, S, slamdam) 
+					S.Knockdown(1 SECONDS) 
+				thrown |= M 
+			for(var/obj/O in thrown)
+				O.take_damage(objdam) // Damage all thrown objects
+		if(T) // if the next tile wont stop the thrown mass from continuing
+			for(var/mob/living/S in thrown)
+				S.Knockdown(1.5 SECONDS)
+				S.Immobilize(1.5 SECONDS)
+			for(var/atom/movable/K in thrown) // to make the mess of things that's being thrown almost look like a normal throw
+				K.SpinAnimation(0.2 SECONDS, 1) 
+				sleep(0.001 SECONDS)
+				K.forceMove(T)
+				if(istype(T, /turf/open/space)) // throw them like normal if it's into space
+					var/atom/throw_target = get_edge_target_turf(K, dir_to_target)
+					K.throw_at(throw_target, 6, 4, user, 3)
+					thrown.Remove(K)
+	drop()
+	return
+
+/*---------------------------------------------------------------
+	end of grapple section
+---------------------------------------------------------------*/
+/*---------------------------------------------------------------
+	start of mop section
+---------------------------------------------------------------*/
+
+/datum/martial_art/buster_style/proc/mop(mob/living/user)
+	var/jumpdistance = 5
+	var/dragdam = 8
+	var/crashdam = 10
+	var/mob/living/B = user
+	var/turf/T = get_step(get_turf(B), B.dir)
+	var/turf/Z = get_turf(B)
```

Comment:
come on
```suggestion
	var/turf/forward_turf = get_step(get_turf(user), user.dir)
	var/turf/current_turf = get_turf(user)
```
i won't change the rest but you know what to do, also why is B needed???

Reply:
iirc the proc wasnt working until i did it but the changes work too